### PR TITLE
Add documentation examples for `unneeded-not`

### PR DESCRIPTION
**Pull request by julthep**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Add documentation examples for `unneeded-not`

Refs #5953

### CODE REVIEW

File: doc/data/messages/u/unneeded-not/good.py [hunk 1]
```diff
@@ -1 +1,2 @@
-# This is a placeholder for correct code for this message.
+if not input():
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/unneeded-not/good.py:1]:
```suggestion
if input():
```

**Review by Pierre-Sassoulas — APPROVED** [review]

File: doc/data/messages/u/unneeded-not/good.py (hunk 1 above)
Comment (julthep) [reply doc/data/messages/u/unneeded-not/good.py]: I thought the good example intend to check for `not input()` but the bad example is mistakenly put duplicated `not'.

Comment (Pierre-Sassoulas) [reply doc/data/messages/u/unneeded-not/good.py]: 👍 I think in this case doing the same thing in both example is possible and match the message name more.

Comment (julthep) [reply doc/data/messages/u/unneeded-not/good.py]: 👍 Agree with you.

**Pull request merged**